.github/shared/src/spec-model.js

```diff
@@ -0,0 +1,316 @@
+    }
+
+    /** @type Swagger[] */
+    const swaggers = [];
+
+    // It's possible for input-file to be a string or an array
+    const inputFiles = Array.isArray(obj["input-file"]) 
+      ? obj["input-file"]
+      : [obj["input-file"]];
+    for (const swaggerPath of inputFiles) {
+      const swagger = await getSwagger(swaggerPath, dirname(readmePath), repoRoot, logger);
+      swaggers.push(swagger);
+    }
+
+    tags.set(tagName, swaggers);
+  }
+
+  const readme = {
+    path: readmePath,
+    content,
+    globalConfig,
+    tags,
+  };
+
+  return readme;
+}
+
+/**
+ * @param {string} swaggerPath
+ * @param {string} readmeFolder
+ * @param {string} repoRoot
+ * @param {import('./logger.js').ILogger} [logger]
+ * @returns {Promise<Swagger>}
+ */
+async function getSwagger(swaggerPath, readmeFolder, repoRoot, logger) {
+  logger?.debug(`getSwagger: ${swaggerPath}, ${readmeFolder}, ${repoRoot}`);
+
+  let normalizedSwaggerPath = swaggerPath;
+  // Ignore uses of "$(this-folder)" in the swagger path. It refers to the 
+  // current folder anyway and can be substituted with "."
+  if (swaggerPath.includes("$(this-folder)")) {
+    normalizedSwaggerPath = swaggerPath.replaceAll("$(this-folder)", ".");
+  }
+
+
+  // Some swagger paths contain backslashes. These should be normalized when 
+  // encountered though the expected format for input-files is forward slashes.
+  if (normalizedSwaggerPath.includes("\\")) {
+    logger?.info(`Found backslash (\\) in swagger path ${swaggerPath}. Replacing with forward slash (/)`);
+    normalizedSwaggerPath = normalizedSwaggerPath.replaceAll("\\", "/");
+  }
+  const fullPath = resolve(repoRoot, readmeFolder, normalizedSwaggerPath);
+  const content = await readFile(fullPath, { encoding: "utf8" });
+
+  return {
+    path: join(readmeFolder, normalizedSwaggerPath),
+    content,
+    refs: await getExternalFileRefs(normalizedSwaggerPath, readmeFolder, repoRoot),
+  };
+}
+
+/**
+ * Reads all levels of external file references in a swagger file and returns a 
+ * map where the key is the referenced swagger file and the value is a set of 
+ * all references that the file depends on, including transitive dependencies.
+ * @param {string} swaggerPath,
+ * @param {string} readmePath
+ * @param {string} repoRoot
+ * @returns {Promise<Map<string, Set<string>>>}
+ */
+async function getExternalFileRefs(swaggerPath, readmePath, repoRoot) {
+  const initialSwagger = resolve(repoRoot, readmePath, swaggerPath);
+  const visited = new Set();
+  const toVisit = new Set([initialSwagger]);
+  const externalFileRefs = new Map();
+
+  while (toVisit.size > 0) {
+    const currentSwagger = /** @type {string} */ (toVisit.values().next().value);
+
+    toVisit.delete(currentSwagger);
+    visited.add(currentSwagger);
+
+    const currentSchema = await $RefParser.resolve(
+      resolve(repoRoot, currentSwagger),
+      { resolve: { http: false } },
+    );
+    const currentRefs = currentSchema
+      .paths("file")
```

Comment:
`currentSchema.paths()` returns a list of all dependencies (including transitive dependencies) of the given swagger file. This list flattens the relationships between referenced swagger files. This means that, with the code in its current configuration, you can't reconstruct the complete dependency graph for a given input swagger file. 

However, for purposes of LintDiff this limitation does not cause problems. It also may not cause problems in other parts of the code so I'm not treating this limitation as requiring resolution at this time.